### PR TITLE
@vuepress-seach / Searchable paths

### PR DESCRIPTION
**Pull request by robsontenorio**
**Summary**

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**The PR fulfills these requirements:**

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome (latest)
- [x] Firefox (latest)
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature
- [x] Related documents have been updated
- [ ] Related tests have been updated

### searchPaths
 - Type: `array`
- Default: []
 Set the array of searchable paths. Default value (a empty array) will search on all paths. Considering you have this structure:
 ```bash
docs/
├── .vuepress/            
│    └── ...
│
├── 1.0/               
│    └── ...
│
└── 2.0/               
     └── ...                       
```
 You can restrict the searching setting with `searchPaths` as `['/1.0/']`. Otherwise,  the default search will return duplications, once you can have similar content between folders `/1.0/` and `/2.0/`.

### CODE REVIEW

**Comment by matrunchyk** [comment]
Please, please, it's very important!

**Comment by chrissm79** [comment]
This would be 💯

File: packages/@vuepress/plugin-search/SearchBox.vue [hunk 1]
```diff
@@ -112,6 +118,23 @@ export default {
...
+
+      searchPaths.forEach((path) => {
+        if (page.path.indexOf(path) !== -1) {
+          searchable = true
+          return
+        }
+      })
+      return searchable
+    },
+
```
Comment (ulivz) [inline packages/@vuepress/plugin-search/SearchBox.vue]: Code here can be simplified as:

```js
isSearchable (page) {
  const searchPaths = SEARCH_PATHS
  // all paths searchables
  if (searchPaths.length === 0) { return true }

  return searchPaths.filter(path => {
    return page.path.indexOf(path) !== -1
  }).length > 0
},
```

**Comment by ulivz** [comment]
So I recommend to name this API as `test`:

- `RegExp`:

```js
module.exports = {
  plugins: [
    ['@vuepress/search', {
      test: /1\.0/
    }]
  ]
}
```

- `Array<RegExp>`:

```js
module.exports = {
  plugins: [
    ['@vuepress/search', {
      test: [ /1\.0/, /2\.0/ ]
    }]
  ]
}
```

- `Function (This should be done in another PR)`

```js
module.exports = {
  plugins: [
    ['@vuepress/search', {
      test: (currentPage, pages) => { }
    }]
  ]
}
```

**Comment by robsontenorio** [comment]
@ulivz Done!

**Pull request merged**